### PR TITLE
redesign ui elements

### PR DESCRIPTION
**Pull request by cmkohnen**
This redesigns UI, and underlying code, around mode switching.
Cut data is now moved to the manipulate category as discussed in https://github.com/SjoerdB93/Graphs/issues/61
It is now no longer possible to have none of these buttons selected, see https://github.com/SjoerdB93/Graphs/issues/62
Reset view button is removed, thinking of improving a lot of actions via keyboard shortcuts in a different pr

It looks like this:
grafik

Feedback appreciated, if it should resemble something like [this](https://gitlab.gnome.org/GNOME/libadwaita/-/issues/212) like @bertob suggested in https://github.com/SjoerdB93/Graphs/issues/62 as discussion around this widget seems to be ongoing and we would have to implement our own version

On the backend side:
I've done a little cleanup and regrouping, to keep related code closer together.
I've also used an enum for the mode and updated all references

### CODE REVIEW

**Comment by SjoerdB93** [comment]
I think this works quite well, I've looked a bit at what the Stack Switcher looks like these days with LibAdwaita, but it's basically the same as these toggles but without the padding in between. Personally I feel like this works just as well UI-wise.

I'm still a bit conflicted on whether the "Normal mouse pointer" mode should be coupled to a "Restore View" action. This would give a quick and easy way to restore the view, while still giving a consistent experience ("Mouse pointer mode" always shows the graphs in a full zoomed-out mode). On the other hand, I'm not 100% sure if having the graph reset view when going into neutral mode is the best course of action, I can also imagine some might find that annoying. And binding the "Restore View" to a keyboard shortcut would indeed get the job done as well (also it's available from the menu). But it feels natural to me to have a quick way in GUI to reset view, and we could just as well bind that to the Restore view..

I'm open to either comments on that.

File: src/main.py [hunk 1]
```diff
@@ -37,6 +38,11 @@ def load_preferences(self):
     def connect_actions(self):
         self.create_action('quit', self.on_quit_action, ['<primary>q'])
         self.create_action('about', self.on_about_action)
+        self.create_action('toggle_sidebar', self.toggle_sidebar, None)
+        self.create_action('mode_none', self.set_mode, None, InteractionMode.NONE)
+        self.create_action('mode_pan', self.set_mode, None, InteractionMode.PAN)
+        self.create_action('mode_zoom', self.set_mode, None, InteractionMode.ZOOM)
+        self.create_action('mode_select', self.set_mode, None, InteractionMode.SELECT)
         self.create_action('preferences', preferences.open_preferences_window, ['<primary>p'], self)
```
Comment (SjoerdB93) [inline src/main.py:46]: We should probably bind some shortcuts to some of these actions, but as said it's probably best to do that in a separate PR

File: src/main.py [hunk 2]
```diff
@@ -88,10 +90,11 @@ def do_activate(self):
         graphs.load_empty(self)
         graphs.disable_clipboard_buttons(self)
         graphs.enable_data_dependent_buttons(self, False)
+        self.set_mode(None, None, InteractionMode.NONE)
         win.maximize()
         win.present()
 
-    def on_about_action(self, _widget, _):
+    def on_about_action(self, widget, shortcutself):
```
Comment (SjoerdB93) [inline src/main.py:97]: I'd go for just shortcut instead of shortcutself

**Review by SjoerdB93 — APPROVED** [review]

**Review by SjoerdB93 — APPROVED** [review]

**Comment by cmkohnen** [comment]
We could use an AdwSplitButton for the view menu und have the primary function be to reset the view. Then there'd still be a quick option to reset the view in UI, but it wouldn't confuse or annoy users

**Comment by SjoerdB93** [comment]
> We could use an AdwSplitButton for the view menu und have the primary function be to reset the view. Then there'd still be a quick option to reset the view in UI, but it wouldn't confuse or annoy users

I think that would be a really nice solution, at least for me that would work well!

**Comment by SjoerdB93** [comment]
This works really well on my machine, and fits nicely in my workflow at least without needing to bind an action to a specific mode (mouse pointer mode in this case). For my part this can be merged.
